pkg/collector/corechecks/network-devices/cisco-sdwan/payload/cedge_interface.go

```diff
@@ -93,38 +93,74 @@ func (itf *CEdgeInterface) Metadata(namespace string) (devicemetadata.InterfaceM
 	}, nil
 }
 
-// IPAddressMetadata returns the metadata for this interface's IP addresses
-func (itf *CEdgeInterface) IPAddressMetadata(namespace string) (devicemetadata.IPAddressMetadata, error) {
+// IPV4AddressMetadata returns the metadata for this interface's IPV4 addresses
+func (itf *CEdgeInterface) IPV4AddressMetadata(namespace string) (*devicemetadata.IPAddressMetadata, error) {
+	if isEmptyCEdgeIP(itf.IPAddress) {
+		return nil, nil
+	}
+
+	index, err := itf.Index()
+	if err != nil {
+		return nil, err
+	}
+
+	ip, err := parseCEdgeIP(itf.IPAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	prefixLen, err := parseMask(itf.Ipv4SubnetMask)
+	if err != nil {
+		return nil, err
+	}
+
+	return &devicemetadata.IPAddressMetadata{
+		InterfaceID: fmt.Sprintf("%s:%s:%d", namespace, itf.VmanageSystemIP, index),
+		IPAddress:   ip,
+		Prefixlen:   prefixLen,
+	}, nil
+}
+
+// IPV6AddressMetadata returns the metadata for this interface's IPV6 addresses
+func (itf *CEdgeInterface) IPV6AddressMetadata(namespace string) (*devicemetadata.IPAddressMetadata, error) {
+	if isEmptyCEdgeIP(itf.IPV6Address) {
+		return nil, nil
+	}
+
 	index, err := itf.Index()
 	if err != nil {
-		return devicemetadata.IPAddressMetadata{}, fmt.Errorf("unable to process cEdge interface %s index : %s", itf.Ifname, err)
+		return nil, err
 	}
 
-	ip, prefiLen, err := parseIPFromCEdgeInterface(itf.IPAddress, itf.Ipv4SubnetMask)
+	ip, err := parseCEdgeIP(itf.IPV6Address)
 	if err != nil {
-		return devicemetadata.IPAddressMetadata{}, fmt.Errorf("unable to process cEdge interface %s IP address : %s", itf.Ifname, err)
+		return nil, err
 	}
 
-	return devicemetadata.IPAddressMetadata{
+	return &devicemetadata.IPAddressMetadata{
```

Comment:
😮  We don't have the subnet mask for IPv6 addresses? I'm surprised, is there any way to fetch it?

Reply:
I don't think so, at least I can't find it anywhere 😕